Automattic/jetpack
# Admin Menu: Mimic existing Calypso menu on Jetpack sites

Fixes (partially) https://github.com/Automattic/wp-calypso/issues/50438

#### Changes proposed in this Pull Request:
Implements some enhancements to improve the UX of the new unified Calypso menu on Jetpack sites:
- Update `Jetpack_Admin_Menu` so we don’t return more items that the ones currently present in the existing navigation.
- Make sure all links point to Calypso (regardless of the link destination toggle). There are a few exception for this, since there are some screens that are not supported by Calypso (Feedback, Customize, Import, Export, WP Admin).

This will be followed up with a Calypso PR that will display an external icon on the WP Admin exceptions while ensuring they will be opened on a new tab: https://github.com/Automattic/wp-calypso/pull/50444

Existing nav | New unified nav (*)
--- | ---
<img width="268" alt="Screen Shot 2021-02-24 at 17 24 10" src="https://user-images.githubusercontent.com/1233880/109032493-00906f80-76c6-11eb-9ff1-03d8a492c899.png"> | <img width="266" alt="Screen Shot 2021-02-24 at 17 26 33" src="https://user-images.githubusercontent.com/1233880/109032512-05552380-76c6-11eb-857d-4d5636cdd6db.png">

(*) I manually expanded all the section to make visible all the submenu. In practice, only one section can be expanded at the same time.

#### Jetpack product discussion
p1HpG7-bcl-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- Spin up a Jetpack site running this branch.
- Go to wordpress.com and switch to that site.
- Make sure the menu items match with what's in the existing navigation (you can switch to that by adding `?disable-nav-unification` to the URL).
- Install some third party plugins and make sure their menus don't appear in the new unified menu.
- (Optional): Test how this looks in combination with https://github.com/Automattic/wp-calypso/pull/50444.

#### Proposed changelog entry for your changes:
N/A.
